dask_expr/_expr.py

```diff
@@ -1615,6 +1619,9 @@ def _simplify_down(self):
         if (
             str(self.frame.columns) == str(self.columns)
             and self._meta.ndim == self.frame._meta.ndim
+            and not (
```

Comment:
I am not sure I understand why this is necessary. Can you elaborate?

Why can't we return self.frame if the columns are the same anyway?

Reply:
My original thinking here was that if `self.frame` is a `BlockwiseIO` expression that can "absorb" projections, then we want to make sure `self.frame` actually absorbs the projection (by applying its own `simplify_up` logic). 

With that said, you are correct that "absorbing" the projection should not really change anything if the first two criteria of this `if` statement are `True`. Therefore, we probably can/should revert this change.  Note that I submitted https://github.com/dask-contrib/dask-expr/pull/267 to do this (where I included a necessary bug fix).